git-link.el

```diff
@@ -342,29 +351,31 @@ return (FILENAME . REVISION) otherwise nil."
 		       (car remotes)))))
 
 (defun git-link--get-region ()
-  (save-restriction
-    (widen)
-    (save-excursion
-      (let* ((use-region (use-region-p))
-             (start (when use-region (region-beginning)))
-             (end   (when use-region (region-end)))
-             (line-start (line-number-at-pos start))
-             line-end)
-        (when use-region
-          ;; Avoid adding an extra blank line to the selection.
-          ;; This happens when point or mark is at the start of the next line.
-          ;;
-          ;; When selection is from bottom to top, exchange point and mark
-          ;; so that the `point' and `(region-end)' are the same.
-          (when (< (point) (mark))
-            (exchange-point-and-mark))
-          (when (= end (line-beginning-position))
-            ;; Go up and avoid the blank line
-            (setq end (1- end)))
-          (setq line-end (line-number-at-pos end))
-          (when (<= line-end line-start)
-            (setq line-end nil)))
-        (list line-start line-end)))))
+  (cond
+   ((git-link--using-magit-blame) (list (oref (magit-blame-chunk-at (point)) :orig-line) nil))
+   (buffer-file-name (save-restriction
```

Comment:
Why are we calling `buffer-file-name` here? This is supposed to return start and end of region.

Reply:
Oh [I see why](https://github.com/sshaw/git-link/pull/60/files#diff-dbf604e34c6900c6747ecbe57dff621aL459). But the check of `buffer-file-name` variable must be done _before_ we attempt to get the region else there will be an error. 

When using magit-blame `buffer-file-name` is defined, so not sure why you removed it (Maybe it's not defined in your version? I know the latest magit made changes to its blame mode).